62.CAD/main.cpp

```diff
@@ -1572,12 +1703,17 @@ class CADApp : public ApplicationBase
 		logicalDevice->waitIdle();
 	}
 
-	double getScreenToWorldRatio(const double4x4& viewProjectionMatrix, uint2 windowSize)
+	float getScreenToWorldRatio(const double4x4& viewProjectionMatrix, uint2 windowSize)
 	{
 		double idx_0_0 = viewProjectionMatrix._r0[0u] * (windowSize.X / 2.0);
 		double idx_1_1 = viewProjectionMatrix._r1[1u] * (windowSize.Y / 2.0);
 		double det_2x2_mat = idx_0_0 * idx_1_1;
-		return core::sqrt(core::abs(det_2x2_mat));
+		return static_cast<float>(core::sqrt(core::abs(det_2x2_mat)));
+	}
+
+	inline float convertToWorldToScreenRatio(float screenToWorldRatio)
+	{
+		return 1.0f / screenToWorldRatio;
```

Comment:
lol what is this function, seem like a general reciprocal.
you can get `const double4x4& viewProjectionMatrix, uint2 windowSize` like the function above and call `getScreenToWorldRatio` internally.

Reply:
done